test/disjointCoarseMallocPool.cpp

```diff
@@ -0,0 +1,610 @@
+    constexpr int allocs_size = 10;
+    void *allocs[allocs_size] = {0};
+    for (int i = 0; i < allocs_size; i++) {
+        ASSERT_EQ(umfCoarseMemoryProviderGetStats(prov).used_size, i * 2 * MB);
+        allocs[i] = umfPoolMalloc(pool, 2 * MB);
+        ASSERT_NE(allocs[i], nullptr);
+    }
+    ASSERT_EQ(umfCoarseMemoryProviderGetStats(prov).used_size, 20 * MB);
+    ASSERT_EQ(umfCoarseMemoryProviderGetStats(prov).alloc_size,
+              init_buffer_size);
+    // there should be no block with the free memory
+    ASSERT_EQ(umfCoarseMemoryProviderGetStats(prov).blocks_num, allocs_size);
+
+    // free all memory
+    for (int i = 0; i < allocs_size; i++) {
+        res = umfPoolFree(pool, allocs[i]);
+        ASSERT_EQ(res, UMF_RESULT_SUCCESS);
+    }
+
+    ASSERT_EQ(umfCoarseMemoryProviderGetStats(prov).blocks_num, 1);
+    ASSERT_EQ(umfCoarseMemoryProviderGetStats(prov).used_size, 0 * MB);
+    ASSERT_EQ(umfCoarseMemoryProviderGetStats(prov).alloc_size,
+              init_buffer_size);
+
+    umfPoolDestroy(pool);
+    umfMemoryProviderDestroy(coarse_memory_provider);
+    umfMemoryProviderDestroy(malloc_memory_provider);
+}
+
+TEST_F(test, disjointCoarseMallocPool_simple1) {
+    umf_memory_provider_handle_t malloc_memory_provider;
+    umfMemoryProviderCreate(&UMF_MALLOC_MEMORY_PROVIDER_OPS, NULL,
+                            &malloc_memory_provider);
+    ASSERT_NE(malloc_memory_provider, nullptr);
+
+    const size_t KB = 1024;
+    const size_t MB = 1024 * KB;
+
+    const size_t init_buffer_size = 20 * MB;
+
+    coarse_memory_provider_params_t coarse_memory_provider_params = {
+        malloc_memory_provider, // upstream_memory_provider
+        true,                   // immediate_init
+        0,
+        init_buffer_size,
+        true,  // trace
+        false, // WA_do_not_free_upstream
+    };
+
+    umf_memory_provider_handle_t coarse_memory_provider;
+    umfMemoryProviderCreate(umfCoarseMemoryProviderOps(),
+                            &coarse_memory_provider_params,
+                            &coarse_memory_provider);
+    ASSERT_NE(coarse_memory_provider, nullptr);
+
+    umf_disjoint_pool_params_t disjoint_memory_pool_params = {};
+    disjoint_memory_pool_params.SlabMinSize = 4096;
+    disjoint_memory_pool_params.MaxPoolableSize = 4096;
+    disjoint_memory_pool_params.Capacity = 4;
+    disjoint_memory_pool_params.MinBucketSize = 64;
+    disjoint_memory_pool_params.PoolTrace = 1;
+
+    umf_memory_pool_handle_t pool;
+    umfPoolCreate(umfDisjointPoolOps(), coarse_memory_provider,
+                  &disjoint_memory_pool_params, UMF_POOL_CREATE_FLAG_NONE,
+                  &pool);
+    ASSERT_NE(pool, nullptr);
+
+    umf_memory_provider_handle_t prov = NULL;
+    umfPoolGetMemoryProvider(pool, &prov);
+    ASSERT_NE(prov, nullptr);
+
+    // test 1
+
+    size_t s1 = 74659 * KB;
+    size_t s2 = 8206 * KB;
+
+    size_t max_alloc_size = 0;
+
+    // s1
+    for (int j = 0; j < 2; j++) {
```

Comment:
magic values: `2`, `6`, please use some consts

Reply:
Fixed in https://github.com/oneapi-src/unified-memory-framework/pull/715